plugin/action/mask/mask.go

```diff
@@ -204,19 +234,27 @@ func getValueNodeList(currentNode *insaneJSON.Node, valueNodes []*insaneJSON.Nod
 func (p *Plugin) Do(event *pipeline.Event) pipeline.ActionResult {
 	root := event.Root.Node
 
+	maskApplied := false
+	locApplied := false
 	p.valueNodes = p.valueNodes[:0]
 	p.valueNodes = getValueNodeList(root, p.valueNodes)
-
 	for _, v := range p.valueNodes {
 		value := v.AsBytes()
 		p.sourceBuf = append(p.sourceBuf[:0], value...)
 		p.maskBuf = append(p.maskBuf[:0], p.sourceBuf...)
 		for _, mask := range p.config.Masks {
-			p.maskBuf = maskValue(p.sourceBuf, p.maskBuf, mask.Re_, mask.Groups, p.logger)
+			p.maskBuf, locApplied = maskValue(p.sourceBuf, p.maskBuf, mask.Re_, mask.Groups)
 			p.sourceBuf = p.maskBuf
+			if locApplied {
+				maskApplied = true
+			}
 		}
 		v.MutateToString(string(p.maskBuf))
 	}
+	if p.logMaskAppeared.Load() && maskApplied {
+		p.counterMetric.Inc()
+		p.logger.Infof("mask appeared to event, output string: %s", string(p.maskBuf))
```

Comment:
p.maskBuf changes on each iteration of loop you will see here only one value from event